Feature/detail view

MurakamiTraining opened the pull request:
・外部ライブラリを追加
　→Alamofire　（HTTP通信用）
　→SwiftSoup　（HTMLの解析用）
　→SDWebImage　（Web画像の取得、キャッシュ管理用）
・記事詳細画面を追加
・RSS一覧でタップされた記事の詳細を表示
・記事掲載サイトのHTMLを解析して必要な情報のみ表示
・記事掲載サイトの画像を取得してトップに表示
・シェア機能
・お気に入り登録、解除機能
　　→（現状UserDefaultに配列としてidを保持）
・その他記事詳細画面追加に伴う、他箇所の修正

MurakamiTraining committed:
・外部ライブラリを追加
　→Alamofire　（HTTP通信用）
　→SwiftSoup　（HTMLの解析用）
　→SDWebImage　（Web画像の取得、キャッシュ管理用）
・記事詳細画面を追加
・RSS一覧でタップされた記事の詳細を表示
・記事掲載サイトのHTMLを解析して必要な情報のみ表示
・記事掲載サイトの画像を取得してトップに表示
・その他記事詳細画面追加に伴う、他箇所の修正

MurakamiTraining committed:
・記事詳細画面に下記機能を追加
　→シェア機能
　→お気に入り登録、解除機能
　　（現状UserDefaultに配列としてidを保持）
・記事詳細取得処理を一部調整

fukuta-t commented:
PRの向き先がdevelopでなくmainなのは何か理由がありますか？
feature-1ブランチがdevelopにマージされているのでそこから切り直す方が良いかと思います

ill-amat commented on training/ClientRSS.swift:
```diff
@@ -6,11 +6,20 @@
 //
 
 import UIKit
+import Foundation
```
[NITS] UIKit で Foundationのimportはできているので import Foundation の記述は不要

MurakamiTraining replied:
該当箇所を削除

ill-amat commented on training/ViewControllerMain.swift:
```diff
@@ -102,6 +102,22 @@ class ViewControllerMain: ViewControllerExtension, UITableViewDelegate, UITableV
+        
+        let viewControllerDetail = moveNextView(storyboardID: .Detail) as? ViewControllerDetail
+        
+        guard viewControllerDetail != nil else {
```
[must] guard文は記述以降、そのスコープ内で値がnilでないことを保証することができます
現状の記述のようにnilでないことを確認する条件式を書く必要はありません
以下のように記述すれば以降detailViewControllerがnildでないことが保証されます
`guard let detailViewController = viewControllerDetail else {
            return
        }`

MurakamiTraining replied:
guardの内容を改めて確認し修正

ill-amat commented on training/ViewControllerMain.swift:
```diff
@@ -102,6 +102,22 @@ class ViewControllerMain: ViewControllerExtension, UITableViewDelegate, UITableV
+            return
+        }
+        
+        viewControllerDetail!.RequestRSSDetail(rssSimple: rssList.items[indexPath.row])
```
[must] 上記で指摘した修正を加えることで強制アンラップを行う必要がなくなります
`detailViewController.RequestRSSDetail(rssSimple: rssList.items[indexPath.row])`

ill-amat commented on training/ViewControllerExtension.swift:
```diff
@@ -17,10 +17,12 @@ class ViewControllerExtension: UIViewController {
         let nextVC = self.storyboard?.instantiateViewController(withIdentifier: storyboardID.rawValue)
         navigationController?.pushViewController(nextVC!, animated: true)
+        
+        return nextVC
```
[must] moveNextViewの役割は指定した画面への遷移かと思います。
遷移先画面の取得を行うのであれば別の関数を定義してください

MurakamiTraining replied:
遷移先画面の取得関数を切り分けして対応

MurakamiTraining committed:
プルリクエストでの指摘点を修正
・不要なImportを削除
・画面遷移時の関数の役割を切り分け
・guard文の間違いを修正

fukuta-t commented on training/ViewControllerDetail.swift:
```diff
@@ -0,0 +1,152 @@
+class ViewControllerDetail: ViewControllerExtension {
+    
+    @IBOutlet weak var rssImageView: UIImageView!
+    
```
1行ごとに改行する必要はないので削って欲しいです

MurakamiTraining replied:
不要な改行を削除

fukuta-t commented on training/ViewControllerDetail.swift:
```diff
@@ -0,0 +1,152 @@
+        SDWebImageManager.shared.loadImage(with: URL(string: rssDetail.image),
+                                           progress: nil,
+                                           completed: {(image, data, error, cache, finished, url) in
+            
```
ここのような不要な改行も削って欲しいです（全体的に）

MurakamiTraining committed:
不要な改行などを削除

MurakamiTraining committed:
guardが意図しない形になっていたため修正
強制 アンラップ箇所を修正

MurakamiTraining committed:
プルリクエスト時の指摘を修正
・無駄な改行、空白の削除
・本ブランチで追加したクラスを命名規則に従う形に修正
・関数に関して命名規則に従う形に修正
・その他全体的な指摘に関してはリファクタブランチを作成して対応
